File: backend/Models/Incident.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace IncidentTracker.Api.Models;

public class Incident
{
    public Guid Id { get; set; }
    
    [Required]
    [StringLength(200)]
    public string Title { get; set; } = string.Empty;
    
    [Required]
    [StringLength(100)]
    public string Service { get; set; } = string.Empty;
    
    [Required]
    public Severity Severity { get; set; }
    
    [Required]
    public Status Status { get; set; }
    
    [StringLength(100)]
    public string? Owner { get; set; }
    
    [StringLength(2000)]
    public string? Summary { get; set; }
    
    public DateTime CreatedAt { get; set; }
    
    public DateTime UpdatedAt { get; set; }
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum Severity
{
    SEV1,
    SEV2,
    SEV3,
    SEV4
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum Status
{
    OPEN,
    MITIGATED,
    RESOLVED
}

File: frontend/src/types/incident.ts
export enum Severity {
  SEV1 = 'SEV1',
  SEV2 = 'SEV2',
  SEV3 = 'SEV3',
  SEV4 = 'SEV4',
}

export enum Status {
  OPEN = 'OPEN',
  MITIGATED = 'MITIGATED',
  RESOLVED = 'RESOLVED',
}

export interface Incident {
  id: string;
  title: string;
  service: string;
  severity: Severity;
  status: Status;
  owner: string | null;
  summary: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateIncidentRequest {
  title: string;
  service: string;
  severity: Severity;
  status: Status;
  owner?: string;
  summary?: string;
}

export interface UpdateIncidentRequest {
  title?: string;
  service?: string;
  severity?: Severity;
  status?: Status;
  owner?: string;
  summary?: string;
}

export interface PagedResponse<T> {
  data: T[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface IncidentQueryParams {
  page?: number;
  pageSize?: number;
  search?: string;
  service?: string;
  severity?: Severity;
  status?: Status;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
}
